operator/controllers/actionsrunner/reconciler.go

```diff
@@ -0,0 +1,298 @@
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	desiredSecret, err := toSecret(wire.DotFiles, &actionsRunner, r.Scheme)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.Patch(ctx, desiredConfigMap, client.Apply, patchOpts...); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.Patch(ctx, desiredSecret, client.Apply, patchOpts...); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.consume(wire); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.source = make(chan event.GenericEvent)
+	r.wires = make(map[types.NamespacedName]*Wire)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&inlocov1alpha1.ActionsRunner{}).
+		Owns(&inlocov1alpha1.ActionsRunnerJob{}).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Secret{}).
+		Owns(&corev1.PersistentVolumeClaim{}).
+		Owns(&batchv1.Job{}).
+		Watches(&source.Channel{Source: r.source}, &handler.EnqueueRequestForObject{}).
+		Complete(r)
+}
+
+func (r *Reconciler) consume(wire *Wire) error {
+	ack, msg, err := wire.Go(r.source)
+	select {
+	case msg := <-msg:
+		switch typ := msg.Type; typ {
+		case facades.MessageTypePipelineAgentJobRequest:
+			return r.onPipelineAgentJobRequest(wire, ack, msg)
+		case facades.MessageTypeJobCancellation:
+			return r.onPipelineAgentJobRequest(wire, ack, msg)
+		case facades.MessageTypeAgentRefresh:
+			return r.onPipelineAgentJobRequest(wire, ack, msg)
+		default:
+			return fmt.Errorf("unknown MessageType(%s)", typ)
+		}
+	case err := <-err:
+		return err
+	default:
+		return nil
+	}
+}
+
+func (r *Reconciler) onPipelineAgentJobRequest(wire *Wire, ack <-chan struct{}, message facades.Message) error {
+	r.Log.Info("On PipelineAgentJobRequest")
+
+	if wire == nil {
+		return errors.New("wire == nil")
+	}
+
+	if message.Type != facades.MessageTypePipelineAgentJobRequest {
+		return fmt.Errorf("message.Type != MessageTypePipelineAgentJobRequest")
+	}
+
+	desiredActionsRunnerJob, err := toActionsRunnerJob(wire.ActionsRunner, r.Scheme)
+	if err != nil {
+		return err
+	}
+	if err := r.Patch(wire.Context, desiredActionsRunnerJob, client.Apply, patchOpts...); err != nil {
+		return err
+	}
+
+	desiredPersistentVolumeClaim, err := toPersistentVolumeClaim(wire.ActionsRunner, desiredActionsRunnerJob, r.Scheme)
+	if err != nil {
+		return err
+	}
+	if err := r.Patch(wire.Context, desiredPersistentVolumeClaim, client.Apply, patchOpts...); err != nil {
+		return err
+	}
+
+	desiredJob, err := toJob(wire.ActionsRunner, desiredActionsRunnerJob, r.Scheme)
+	if err != nil {
+		return err
+	}
+	if err := r.Patch(wire.Context, desiredJob, client.Apply, patchOpts...); err != nil {
+		return err
+	}
+
+	key := fmt.Sprintf("%s/%s", desiredActionsRunnerJob.GetNamespace(), desiredActionsRunnerJob.GetName())
+	r.waiting.Store(key, nil)
+	go r.waitForJob(wire.Context, desiredActionsRunnerJob, ack)
+
+	return nil
+}
+
+// TODO
```

Comment:
Still work to be done here? You may specify as `TODO: description`

Reply:
I'll remove the TODO 😬